packages/api/src/hooks/useP2PAdvertList.ts

```diff
@@ -1,12 +1,12 @@
 import React from 'react';
-import { useInfiniteQuery } from '@deriv/api';
+import useInfiniteQuery from '../useInfiniteQuery';
 
 /**
  * This custom hook returns available adverts for use with 'p2p_order_create' by calling 'p2p_advert_list' endpoint
  */
 const useP2PAdvertList = (
     payload?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_advert_list'>>[1]>['payload'],
-    config?: { enabled?: boolean }
+    config?: NonNullable<Parameters<typeof useInfiniteQuery<'p2p_advert_list'>>[1]>['options']
```

Comment:
Is it a good idea to create a function, like `getP2pAdvertList()`, which takes payload and config as inputs and returns them from the hook? I'm bringing this up because we previously discussed about avoiding passing arguments in hook calls.

Reply:
For now since this hook is being used in P2P package and other required hooks for handling the payload/dependencies inside this hook is not yet completed, we still need these arguments in order for the P2P package to pass its dependencies.
I'll add a TODO to refactor this hook once the other required hooks are implemented and migrated to API package